src/pipeline.rs

```diff
@@ -382,3 +430,98 @@ pub(crate) fn prepare_view_targets(
         }
     }
 }
+
+#[derive(Resource, Clone)]
+pub struct PointCloudPlaybackControl {
+    pub playing: bool,
+    pub speed: f32,
+}
+impl Default for PointCloudPlaybackControl {
+    fn default() -> Self {
+        Self {
+            playing: true,
+            speed: 5.0,
+        }
+    }
+}
+
+impl ExtractResource for PointCloudPlaybackControl {
+    type Source = Self;
+
+    /// Defines how the resource is transferred into the "render world".
+    fn extract_resource(source: &Self::Source) -> Self {
+        source.clone()
+    }
+}
+
+pub fn prepare_animated_assets(
+    queue: Res<RenderQueue>,
+    mut assets: ResMut<RenderAssets<PointCloudAsset>>,
+    playback: ResMut<PointCloudPlaybackControl>,
+    time: Res<Time>,
+) {
+    if !playback.playing {
+        return;
+    }
+    for (_handle, asset) in assets.iter_mut() {
+        if let Some(animation_buffer) = asset.animation_buffer.as_mut() {
+            let mut view = vec![0.0; asset.num_points as usize * 3];
```

Comment:
Why not make this a `Vec<Vec3>`?

Reply:
This being Vec<f32> allows us to directly convert it into `&[u8]` without worrying about alignment. Ik Vec3 has alignment=4 but when dealing with casts like this I would opt to use more primitive types.